qualification/Jenkinsfile

```diff
@@ -73,18 +73,28 @@ pipeline {
          script {
            sh 'pip install -r qualification/requirements.txt'
            sh 'pip install --no-deps ".[qualification]" && pip check'
-           try {
-             sh 'spead2_net_raw pytest -v -c qualification/pytest-jenkins.ini qualification --image-override katgpucbf:harbor.sdp.kat.ac.za/cbf/katgpucbf:latest'
-           }
-           finally {
-             COMMIT_ID = sh(script: 'qualification/report/generate_pdf.py report.json report.pdf -c', returnStdout: true)
-             sh(script: 'qualification/report/generate_pdf.py report.json procedure.pdf --generate-procedure-doc')
-             publishHTML(target: [keepAll: true, reportName: 'Qualification Test Report', reportDir: '', reportFiles: 'report.pdf, report.json', reportTitles: 'report.pdf, report.json'])
-             publishHTML(target: [keepAll: true, reportName: 'Qualification Test Procedure', reportDir: '', reportFiles: 'procedure.pdf', reportTitles: 'procedure.pdf'])
-             currentBuild.displayName = "#${BUILD_NUMBER} (${COMMIT_ID})"
-           }
+           sh 'spead2_net_raw pytest -v -c qualification/pytest-jenkins.ini qualification --image-override katgpucbf:harbor.sdp.kat.ac.za/cbf/katgpucbf:latest --junitxml=result.xml'
```

Comment:
Can we have this line of code in a try block to catch errors with the starting up of a correlator, or bugs in the running of the qualification tests?

Reply:
The issue that I have is that, unless there's some Jenkins magic that I don't understand, you catch all problems, whether it's a test that works but doesn't meet spec, or an actual bug in the process somehow.

Currently this is what's happening:
![image](https://user-images.githubusercontent.com/10864691/221161773-a872ed05-ede1-4862-bb1e-af17d3196d85.png)

which is exactly what I want. The tests are failing, so it lets us know about it, and the report generation isn't working, so that block is failing as well.

@amishatishpatel would you be able to weigh in?